A84-xds-authority-literal-rewrite.md

```diff
@@ -0,0 +1,104 @@
+Title
+----
+* Author(s): Zhiyan Foo
+* Approver: a11r
+* Status: Draft
+* Implemented in: TBD
+* Last updated: 2024-11-06
+* Discussion at: TBD (filled after thread exists)
+
+## Abstract
+
+Propose extending gRPC's xDS client to honour `host_rewrite_literal` in route configuration for
+authority header rewriting when `trusted_xds_server` is set.
+
+## Background
+
+gRPC xDS currently supports `auto_host_rewrite` as per [gRFC A81][A81] but lacks `host_rewrite_literal`
+support, useful when a single cluster targets a reverse proxy routing based on the authority
+header.
+```
+            +------------------+
+            |  gRPC xDS Client |  <- (RDS configuration sets `host_rewrite_literal` to
+            +------------------+      target one of Svc A/B/C)
+                     |
+                     |    <- (gRPC xDS client reuses the same CDS target for all connections to
+                     |        the proxy)
+                 +---------+
+                 |  Proxy  |
+                 +---------+
+                      |   <- (Proxy routing logic based on Authority header)
+      .---------------+--------------.
+     /                |               \
++--------+       +--------+       +--------+
+| Svc A  |       | Svc B  |       | Svc C  |
++--------+       +--------+       +--------+
+```
+
+### Related Proposals:
+* [gRFC A29: xDS-Based Security for gRPC Clients and Servers][A29]
+* [gRFC A81: xDS Authority Rewriting][A81]
+* [gRFC A86: xDS-Based HTTP CONNECT][A86]
+
+
+## Proposal
+
+Implement [`host_rewrite_literal`][route_action] feature in gRPC xDS client, enabling explicit
+authority header rewrites. This will be conditioned on `trusted_xds_server` as described in
+[gRFC A81][A81].
+
+### xDS Resource Validation
+
+When validating an RDS resource, if the `trusted_xds_server` server option is present for this
+xDS server in the bootstrap config, we will look at the `RouteAction.host_rewrite_literal`
```

Comment:
We can check to see precdence and see what to do if multiple host rewrite fields are set.